File: binary_search/java/src/main/java/com/okayhak/BinarySearch.java
package com.okayhak;

public class BinarySearch {

    public static int binarySearch(int[] array, int target) {
        if (array.length == 0) {
            return -1;
        }

        int low = 0;
        int high = array.length - 1;
        while (low <= high) {
            int mid = (low + high) / 2;

            if (target == array[mid]) {
                return mid;
            }
            if (target < array[mid]) {
                high = mid - 1;
            }
            if (target > array[mid]) {
                low = mid + 1;
            }
        }
        return -1;
    }
}

File: binary_search/go/binary_search.go
package binary_search

func BinarySearch(array []int, target int) int {
	low, high := 0, len(array)-1

	for low <= high {
		mid := (low + high) / 2

		if target == array[mid] {
			return mid
		}

		if target < array[mid] {
			high = mid - 1
		}

		if target > array[mid] {
			low = mid + 1
		}

	}
	return -1
}
